eng/pipelines/templates/steps/targeting-string-resolve.yml

```diff
@@ -14,5 +14,12 @@ steps:
 
       Write-Host "Setting TargetingString to $setting"
       Write-Host "##vso[task.setvariable variable=TargetingString]$setting"
+
+      Write-Host "Build reason: $(Build.Reason)"
+      if ("$(Build.Reason)" -eq 'PullRequest') {
```

Comment:
Let's remove this. Handle the build-system-specific values of `Build.Reason` in the function `public_ci()`.

If you look at `in_ci()` what we do is return a specific non-zero number for each system. So we _can_ just check non-falsey for `are we in CI`, but also have the detail if we need it.

Please do the same in `in_public()`.

you should be able to check for the presence of `BUILD_REASON` for azure devops, and `GITHUB_EVENT_NAME ` in gh-actions.

according to chatgpt
```
As of my last knowledge update in January 2022, the possible values for GITHUB_EVENT_NAME in GitHub Actions include:

push: Triggered on push events, such as when commits are pushed to a repository branch.
pull_request: Triggered when a pull request is opened, synchronized, or closed.
release: Triggered when a release is created or edited.
schedule: Triggered on a schedule defined in the workflow configuration.
workflow_run: Triggered when a workflow run is requested or completed.
repository_dispatch: Triggered when a repository_dispatch event is received.
```

You just need to recognize `pull_request`.

Doing the above will get rid of the arbitrary variable set + arbitrary variable check -> direct to what the system actually uses.

Reply:
Done, this is a much better idea